modules/yarpc/yarpc.go

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package yarpc
+
+import (
+	"go.uber.org/fx"
+	"go.uber.org/fx/config"
+
+	"go.uber.org/yarpc"
+	"go.uber.org/yarpc/api/transport"
+	"go.uber.org/yarpc/transport/http"
+	"go.uber.org/yarpc/transport/tchannel"
+	yconfig "go.uber.org/yarpc/x/config"
+
+	"github.com/uber-go/tally"
+	"go.uber.org/zap"
+)
+
+// Module represents a collection of YARPC components.
+type Module struct {
+	l           *zap.Logger
+	handlerCtor fx.Component
+	d           *yarpc.Dispatcher
+}
+
+// Transports represent a collection Procedures that will be registered with a dispatcher.
+type Transports struct {
```

Comment:
We can have a constructor for it, not in this PR, but I don't see any problem adding it.
Middleware could also be populated from the yarpc's `Configurator.LoadConfig`.